cmd/single.go

```diff
@@ -37,15 +38,21 @@ Beware! May take days to complete.`,
 
 		// Initiate a channel of repositories.
 		repositories := make(chan crawler.Repository, 100)
+		// Prepare WaitGroup.
+		var wg sync.WaitGroup
 
 		// Process each domain service.
 		for _, domain := range domains {
 			if domain.Id == domainID {
-				go crawler.ProcessDomain(domain, repositories)
+				wg.Add(1)
+				go crawler.ProcessDomain(domain, repositories, &wg)
 			}
 		}
 
+		// Goroutine that constatly check if all the processes are terminated.
+		go crawler.WaitingLoop(repositories, &wg)
+
 		// Process the repositories in order to retrieve publiccode.yml.
-		crawler.ProcessRepositories(repositories)
+		crawler.ProcessRepositories(repositories, &wg)
```

Comment:
can we start `crawler.ProcessRepositories` in a goroutine and move here the code that wait and close the channel?

Reply:
Ok. Moved crawler.ProcessRepositories in a goroutine.
The code of WaitingLoop should be extracted from the function?